avocado/plugins/pretest/pre_test.py

```diff
@@ -0,0 +1,31 @@
+# This program is free software; you can redistribute it and/or modify
```

Comment:
Having this file in `avocado/plugins/pretest` is rather confusing...  AFAICT the code here is a helper to be used by actual plugins, while this is not a plugin itself.  This differs, for instance, from `avocado/plugins/spawners`, `avocado/plugins/runners`, etc.

It's fine to put common code used by plugins in `avocado/core`.